sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -196,7 +196,7 @@ private boolean validateFields(List<String> fields) {
             return true;
         }
         for (String field: fields) {
-            if (field.contains("\\.")) {
+            if (field.contains(".")) {
```

Comment:
To me this is a separate commit, "Fix bug in validating dotted projection fields", as it fixes an existing bug.